test/unit/config_test.rb

```diff
@@ -0,0 +1,417 @@
+
+require 'helper'
+require 'stringio'
+
+require 'inspec/config'
+
+describe 'Inspec::Config' do
+
+  # ========================================================================== #
+  #                                Constructor
+  # ========================================================================== #
+  describe 'the constructor' do
+    describe 'when no args are provided' do
+      it 'should initialize properly' do
+        cfg = Inspec::Config.new
+        cfg.must_respond_to :final_options
+      end
+    end
+
+    describe 'when CLI args are provided' do
+      it 'should initialize properly' do
+        cfg = Inspec::Config.new({color: true, log_level: 'warn'})
+        cfg.must_respond_to :final_options
+      end
+    end
+
+    # TODO: add test for reading from default config path
+
+  end
+
+  # ========================================================================== #
+  #                              File Validation
+  # ========================================================================== #
+  describe 'when validating a file' do
+    let(:cfg) { Inspec::Config.new({}, cfg_io) }
+    let(:cfg_io) { StringIO.new(ConfigTestHelper.fixture(fixture_name)) }
+    let(:seen_fields) { cfg.final_options.keys.sort }
+
+    describe 'when the file is a legacy file' do
+      let(:fixture_name) { 'legacy' }
+      it 'should read the file successfully' do
+        expected = ['color', 'reporter', 'target_id', 'type'].sort
+        seen_fields.must_equal expected
+      end
+    end
+
+    describe 'when the file is a valid v1.1 file' do
+      let(:fixture_name) { 'basic' }
+      it 'should read the file successfully' do
+        expected = ['create_lockfile', 'reporter', 'type'].sort
+        seen_fields.must_equal expected
+      end
+    end
+
+    describe 'when the file is minimal' do
+      let(:fixture_name) { 'minimal' }
+      it 'should read the file successfully' do
+        expected = ['reporter', 'type'].sort
+        seen_fields.must_equal expected
+      end
+    end
+
+    describe 'when the file has malformed json' do
+      let(:fixture_name) { 'malformed_json' }
+      it 'should throw an exception' do
+        ex = proc { cfg }.must_raise(Inspec::ConfigError::MalformedJson)
+        ex.message.must_include 'Failed to load JSON config'
+        ex.message.must_include 'unexpected token'
+        ex.message.must_include 'version'
```

Comment:
I greatly prefer to check for individual elements of an UX message on an element-by-element basis. Rather than altering the test to only match on one regex (that could fail for any number of reasons, and will only grow in complexity over time), checking for one thing at a time make it obvious which piece is missing.

A comment with example output would be informative though - I'll add that.